src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt

```diff
@@ -407,40 +400,57 @@ internal class TestCheckRuleTask : AbstractCheckTaskTest() {
-                            "C" to VerificationStatus.NA,
-                            "D" to VerificationStatus.NA,
-                            "A" to VerificationStatus.FAILED,
-                            "B" to VerificationStatus.FAILED
-                    )
+            val actualLegs = assertVerificationsLegs(verificationEvent)
+
+            val expectedLegs = listOf(
+                    VerificationEntry().apply {
+                        fields = mapOf(
+                                "A" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "B" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "C" to VerificationEntry().apply { status = VerificationStatus.FAILED },
+                                "D" to VerificationEntry().apply { status = VerificationStatus.FAILED }
+                        )
+                    },
+                    VerificationEntry().apply {
+                        fields = mapOf(
+                                "C" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "D" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "A" to VerificationEntry().apply { status = VerificationStatus.FAILED },
+                                "B" to VerificationEntry().apply { status = VerificationStatus.FAILED }
+                        )
+                    }
             )
 
-            expectedLegs.forEach { (leg, verificationEntryByField) ->
-                val actualLeg = actualLegs[leg].fields
-                assertNotNull(actualLeg) { "The validation event does not contain the expected leg" }
-                verificationEntryByField.forEach { (field, status) ->
-                    val expectedVerificationEntry = actualLeg[field]
-                    assertNotNull(expectedVerificationEntry) { "Actual leg does not contain the expected field" }
-                    assertEquals(status, expectedVerificationEntry.status)
-                }
-            }
+            assertVerificationEntriesLegsByStatus(expectedLegs, actualLegs)
         })
     }
 
 
+    private fun extractVerificationFromEvent(verificationEvent: Event): Verification? {
+        return jacksonObjectMapper().readValue<List<Verification>>(verificationEvent.body.toByteArray()).firstOrNull()
+    }
+
+    private fun assertVerificationsLegs(verificationEvent: Event): List<VerificationEntry> {
+        val verification = extractVerificationFromEvent(verificationEvent)
+        assertNotNull(verification) { "Verification event does not contain the verification" }
+        val actualLegs = verification.fields["legs"]?.fields?.values?.toList()
+        assertNotNull(actualLegs) { "Actual legs is missed" }
+        return actualLegs
+    }
+
+    private fun assertVerificationEntriesLegsByStatus(expectedLegs: List<VerificationEntry>, actualLegs: List<VerificationEntry>) {
```

Comment:
I think it will be better to compare whole verifications here instead of VerificationEntry's